src/main/java/hudson/plugins/ec2/EC2SpotSlave.java

```diff
@@ -94,17 +95,24 @@ public void terminate() {
                                     */
                                     LOGGER.info("EC2 instance already terminated: " + instanceId);
                                 } else {
+                                    // check to see if there is a dynamic keypair associated with this instance, and if so, clean it up
+                                    try {
+                                        cleanupSshKeyPairIfNeeded();
+                                    } catch (AmazonClientException | IOException e) {
+                                        LOGGER.info(() -> "unable to remove instance keypair for : " + instanceId);
+                                    }
+                                    // send request to terminate instance
                                     TerminateInstancesRequest request = new TerminateInstancesRequest(Collections.singletonList(instanceId));
                                     try {
                                         ec2.terminateInstances(request);
-                                        LOGGER.info("Terminated EC2 instance (terminated): " + instanceId);
+                                        LOGGER.info("Terminated EC2 spot instance (terminated): " + instanceId);
                                     } catch (AmazonClientException e) {
                                         // Spot request is no longer valid
                                         LOGGER.log(Level.WARNING, "Failed to terminate the Spot instance: " + instanceId, e);
                                     }
                                 }
                             }
-                        } catch (Exception e) {
+                        } catch (AmazonClientException e) {
```

Comment:
Apparently unnecessary change:

```suggestion
                        } catch (Exception e) {
```